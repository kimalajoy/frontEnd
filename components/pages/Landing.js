import React from 'react';
import { View, Image } from 'react-native';

import styles from '../../styles/styles';
import logo from '../../assets/logo.png'
import { OpacityButton } from '../buttons/OpacityButton'

const landingButtonStyle = [styles.landingButtons, styles.borderRadius, styles.boxShadow]
const landingButtonTextStyle = styles.textLight

export default function Landing({ navigation }) {
  return (
    <View style={ styles.container }>
      <Image
        source={logo}
        style={ styles.logo }
        testID={'Landing.Image'}/>
      <OpacityButton
        text={ 'Search Plants' }
        pressFunction={ () => navigation.navigate('TabNavigation', { screen: 'Search'}) }
        textStyle={ landingButtonTextStyle }
        buttonStyle={ landingButtonStyle }/>
      <OpacityButton
        text={ 'My Garden' }
        pressFunction={ () => navigation.navigate('TabNavigation', { screen: 'Garden'}) }
        textStyle={ landingButtonTextStyle }
        buttonStyle={ landingButtonStyle }/>
    </View>
  );
}
