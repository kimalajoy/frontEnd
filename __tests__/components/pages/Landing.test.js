import React from 'react';
import { render, cleanup, fireEvent } from 'react-native-testing-library';

import Landing from '../../../components/pages/Landing';

describe("Landing Page / Home Screen", () => {
  let landing;

  beforeEach(() => {
    landing = render(<Landing />)
  });

  afterEach(cleanup);

  test('it displays the logo', () => {
    const image = landing.getByTestId("Landing.Image")

    expect(image.props.source.testUri).toBe('../../../assets/logo.png')
    expect(image.type).toBe('Image')
  })

  test('renders "Search Plants" button', () => {
    const searchButton = landing.queryByText('Search Plants')

    expect(searchButton).toBeTruthy()
  })

  test('renders "My Garden" button', () => {
    const gardenButton = landing.queryByText('My Garden')

    expect(gardenButton).toBeTruthy()
  })

  test("it renders home page as expected", () => {
    expect(landing.toJSON()).toMatchSnapshot();
  })
})
