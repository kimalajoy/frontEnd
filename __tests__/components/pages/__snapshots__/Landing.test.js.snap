// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page / Home Screen it renders home page as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fffefe",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 300,
        "width": 300,
      }
    }
    testID="Landing.Image"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#5DA25B",
        "borderRadius": 5,
        "elevation": 15,
        "margin": 10,
        "opacity": 1,
        "padding": 15,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 7,
          "width": 0,
        },
        "shadowOpacity": 0.43,
        "shadowRadius": 9.51,
        "width": 150,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fafafa",
          "fontSize": 19,
        }
      }
    >
      Search Plants
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#5DA25B",
        "borderRadius": 5,
        "elevation": 15,
        "margin": 10,
        "opacity": 1,
        "padding": 15,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 7,
          "width": 0,
        },
        "shadowOpacity": 0.43,
        "shadowRadius": 9.51,
        "width": 150,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fafafa",
          "fontSize": 19,
        }
      }
    >
      My Garden
    </Text>
  </View>
</View>
`;
